bindings/java/mongocrypt/.gitignore

```diff
@@ -0,0 +1,35 @@
+*~
```

Comment:
I didn't know you could nest .gitignores. It's not a convention I've noticed before. It probably makes more sense to add these to the top level .gitignore - rather than here.

Reply:
You can.  I think it makes sense to keep the Java-specific ones here.

Reply:
Ack